modules/replicator/src/main/java/org/apache/ignite/internal/replicator/Replica.java

```diff
@@ -178,7 +178,7 @@ public CompletableFuture<ReplicaResult> processRequest(ReplicaRequest request, S
                             executor
                     )
                     .thenComposeAsync(unused -> listener.invoke(request, senderId), executor);
-        } else if (request instanceof EmptyPrimaryReplicaRequest) {
+        } else if (request instanceof WaitReplicaStateMessage) {
             return completedFuture(new ReplicaResult(null, null));
```

Comment:
What's that? Why do we return ReplicaResult(null, null) as a response for WaitReplicaStateMessage?

Reply:
We are only interested in successful completion of `WaitReplicaStateMessage` and do not need any additiona result, so we use `ReplicaResult(null, null)`. I see that this pattern is used in the codebase, but maybe it is wrong pattern, please correct me if needed

Reply:
I'm wondering why we have two WaitReplicaStateMessage processing blocks.
```
        if (!waitForActualStateFuture.isDone() && request instanceof PrimaryReplicaRequest) {
            var targetPrimaryReq = (PrimaryReplicaRequest) request;

            if (request instanceof WaitReplicaStateMessage) {
```
and
```
} else if (request instanceof WaitReplicaStateMessage) {
```

Reply:
I guess it is because the first block is for the first call of `Replica#processRequest`, when we haven't call `waitForActualState` ever, and the second block is for the situations, when we still receive `WaitReplicaStateMessage`, but `waitForActualState` has already been called, so we don't need to call it again

Reply:
I'd say that we still should have one block and detect whether it's first or repetitive call within it.

Reply:
done